### PR TITLE
patch: fix estimator kwargs in signature

### LINKED ISSUE #692: [BUG] regression in grouped estimator
I spotted that an image in our docs no longer gives the right demo. 

CleanShot 2024-07-24 at 12 12 43

It is the one that relates to the grouped estimator that demos the decay estimator. Before, the decayed line would be different than the normally grouped one.

It seems that when we copy the decay estimators internally that we somehow drop `self.decay_kwargs`.

### PR DESCRIPTION
**Pull request by FBruzzesi**
# Description

Hotfix to properly cloning kwargs in estimator `__init__`. This is both a bug fix and a breaking change!

Fixes #692 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines (ruff)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (also to the readme.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added tests to check whether the new feature adheres to the sklearn convention
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/_scripts/meta-models.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def plot_model(model):
     ("datagrab", FeatureUnion([
          ("discrete", Pipeline([
              ("grab", ColumnSelector("diet")),
-             ("encode", OneHotEncoder(categories="auto", sparse=False))
+             ("encode", OneHotEncoder(categories="auto"))
```
Comment (FBruzzesi) [inline docs/_scripts/meta-models.py:135]: kwarg name changed in one of the latest versions, unrelated to PR

File: docs/_static/meta-models/decay-model.png
Comment (FBruzzesi) [inline docs/_static/meta-models/decay-model.png:1]: @koaning there you go! 😇🙌🏼

**Review by koaning — APPROVED** [review]
LGTM! It always surprises me how small edits can suddenly demand a change in so many files 😅

**Comment by FBruzzesi** [comment]
> It always surprises me how small edits can suddenly demand a change in so many files 😅

Wait until a tiny tiny change to support pyarrow via narwhals will become a rework 🙈 

As soon as we make a new Narwhals release I should be able to open the PR, yet I would like to wait for [pyarrow shift PR](https://github.com/narwhals-dev/narwhals/pull/590) to be merged so that also `add_lags` should be able to operate with pyarrow tables

**Pull request merged**